mobile/src/main/java/rs/readahead/washington/mobile/data/nextcloud/NextCloudRepositoryImp.kt

```diff
@@ -0,0 +1,308 @@
+package rs.readahead.washington.mobile.data.nextcloud
+
+import android.content.Context
+import android.net.Uri
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import android.webkit.MimeTypeMap
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.OwnCloudClientFactory
+import com.owncloud.android.lib.common.OwnCloudCredentialsFactory
+import com.owncloud.android.lib.common.UserInfo
+import com.owncloud.android.lib.common.network.OnDatatransferProgressListener
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.files.ChunkedFileUploadRemoteOperation
+import com.owncloud.android.lib.resources.files.CreateFolderRemoteOperation
+import com.owncloud.android.lib.resources.files.ReadFolderRemoteOperation
+import com.owncloud.android.lib.resources.files.UploadFileRemoteOperation
+import com.owncloud.android.lib.resources.status.GetCapabilitiesRemoteOperation
+import com.owncloud.android.lib.resources.users.GetUserInfoRemoteOperation
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Flowable
+import io.reactivex.FlowableEmitter
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import rs.readahead.washington.mobile.MyApplication
+import rs.readahead.washington.mobile.domain.entity.UploadProgressInfo
+import rs.readahead.washington.mobile.domain.entity.collect.FormMediaFile
+import rs.readahead.washington.mobile.domain.repository.nextcloud.NextCloudRepository
+import timber.log.Timber
+import java.io.File
+import java.util.UUID
+
+
+class NextCloudRepositoryImp(private val context: Context) : NextCloudRepository {
+
+    override fun validateServerUrl(serverUrl: String): Single<Boolean> {
+        return Single.create { emitter ->
+            try {
+                val uri = Uri.parse(serverUrl)
+                val client = OwnCloudClientFactory.createOwnCloudClient(uri, context, false)
+
+                val getCapabilitiesOperation = GetCapabilitiesRemoteOperation()
+                getCapabilitiesOperation.execute(
+                    client, { _, result ->
+
+                        result?.let {
+                            if (it.isSuccess) {
+                                emitter.onSuccess(true)
+                            } else {
+                                emitter.onError(result.exception)
+                            }
+                        } ?: emitter.onSuccess(false)
+                    }, Handler(Looper.getMainLooper())
+                )
+
+            } catch (e: Exception) {
```

Comment:
catch block handles all exceptions, I think it could be more specific. If possible, catch specific exceptions               (MalformedURLException) to better understand the type of failure.

Reply:
I am not sure if it is resolved now, please recheck